call_rag.py

```diff
@@ -45,9 +45,11 @@ def messages_to_prompt(messages):
     )
 
 
-def augment(index, prompt):
+def augment(index, prompt, code):
   # bge-base embedding model
     settings_embed()
     query_engine = index.as_query_engine()
-    response = query_engine.query(prompt)
-    return str(response)
+    response = query_engine.query("Hint for what the next line should be in the given in-progress solution: " + code + " to answer the problem: " + prompt + "Put response in the format of '### Hint: '. Type the code plain text: do not use '''dafny to indicate code. ")
```

Comment:
You could let it use code quotes and filter them out. Others do that, maybe reflection.